pkg/scm/git/git.go

```diff
@@ -196,8 +196,24 @@ func ParseURL(source string) (*url.URL, error) {
 }
 
 func useCopy(protoSpecified bool, source string) bool {
-	glog.V(4).Infof("useCopy proto spec %v local git repo %v has git bin %v", protoSpecified, isLocalGitRepository(source), hasGitBinary())
-	return !isLocalGitRepository(source) || !hasGitBinary()
+	isEmpty := isEmptyGitRepository(source)
+	isLocal := isLocalGitRepository(source)
+	hasGit := hasGitBinary()
```

Comment:
you can optimize this further i think by lazily initialized the variables as needed inside the isLocal if block

eg if it's not a local git repo, you don't care if "hasGit" or "isEmpty" are true.